lib/ex_pesa/Jenga/jenga_base.ex

```diff
@@ -97,6 +97,44 @@ defmodule ExPesa.Jenga.JengaBase do
     end
   end
 
+  @doc """
+  performs `GET` requests.
+  `get_request/2` accepts, parameters encoded url and headers
+  `get_request/3` accepsts, url, body as map and headers
+  """
+  @spec get_request(bitstring(), keyword()) :: {:ok, map()} | {:error, bitstring()}
+  def get_request(url, headers) when is_list(headers), do: get_request(url, nil, headers)
+
+  @spec get_request(bitstring(), map(), keyword()) :: {:ok, map()} | {:error, bitstring()}
+  def get_request(url, body, headers \\ [])
+
+  def get_request(url, body, headers) do
+    # anonymous function get./1 to parse body if provided
+    get = fn client ->
+      opts = [adapter: [recv_timeout: 30_000]]
```

Comment:
you wanted to set the recv_timeout on Tesla?

Reply:
yeah, to avoid repeating the `opts` when calling Tesla.get below.

Reply:
@r-coh Just repeat it for now, we will optimise later

Reply:
@lenileiro alright.

Reply:
there's an option to pass it with the configs but the implantation i did it was somehow not working, then about making headers not optional, i think we can just make it optional , we;re not sure of the nature og all get requests we'll meet